test/docs.js

```diff
@@ -30,6 +35,37 @@ describe('Docs', function() {
               }
             }
           }
+        },
+        '@apostrophecms/migration': {
+          extendMethods(self) {
+            return {
+              add(_super, name, migrationFn, options) {
+                migrations.addList.push(name);
+                migrationFn();
+                return _super(name, migrationFn, options);
+              },
+              eachDoc(_super, criteria, limit, iterator) {
+                if (criteria.cacheInvalidatedAt) {
+                  const _updateOne = self.apos.doc.db.updateOne.bind(self.apos.doc.db);
+                  self.apos.doc.db.updateOne = (filter, update, options) => {
```

Comment:
This is a really big override of every mongo update call to add migrations. There is definitely no need for that. Please reach out if you have questions after reviewing some of the regular migrations in apostrophe core.